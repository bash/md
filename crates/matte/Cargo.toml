[package]
name = "matte"
version = "0.1.0"
edition = "2021"

[dependencies]
anstyle.workspace = true
bat.workspace = true
fmtastic.workspace = true
hostname.workspace = true
matte-language-names = { path = "../matte-language-names" }
pulldown-cmark.workspace = true
smallvec.workspace = true
trait-set.workspace = true
unicode-linebreak-chunked.workspace = true
unicode-width.workspace = true
url.workspace = true

[dev-dependencies]
insta = { workspace = true, features = ["glob"] }
serde = { workspace = true, features = ["derive"] }
toml.workspace = true
