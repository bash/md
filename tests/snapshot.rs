use insta::{assert_snapshot, glob};
use matte::{render, supported_parser_options, Options};
use pulldown_cmark::Parser;
use std::fs::read_to_string;

#[test]
fn test_snippets() {
    glob!("snippets/*.md", |path| {
        let markdown = read_to_string(path).unwrap();
        let rendered = render_to_string(&markdown);
        assert_snapshot!(rendered);
    })
}

fn render_to_string(input: &str) -> String {
    let parser = Parser::new_ext(input, supported_parser_options());
    let mut buffer = Vec::new();
    let options = Options::plain_text(120);
    render(parser, &mut buffer, options).unwrap();
    String::from_utf8(buffer).unwrap()
}
