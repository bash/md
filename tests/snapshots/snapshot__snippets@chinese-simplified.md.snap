---
source: tests/snapshot.rs
expression: rendered
input_file: tests/snippets/chinese-simplified.md
---
[1m[4m[32m[0m[1m[4m[32mChinese Simplified[0m

An example to see how the text wrapping is handled:[0m

тФГ[0m [0mцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМ[0m
тФГ[0m [0mц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝М[0m
тФГ[0m [0mхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣а[0m
тФГ[0m [0mх╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдз[0m
тФГ[0m [0mхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦ[0m
тФГ[0m [0mхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзР[0m
тФГ[0m [0mхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒Й[0m
тФГ[0m [0mшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПС[0m
тФГ[0m [0mшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖[0m
тФГ[0m [0mшпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнж[0m
тФГ[0m [0mхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜шпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВцИСхзРхзРхЬихдЦхЫ╜[0m
тФГ[0m [0mшпнхдзхнжхнжф╣ах╛╖шпня╝МхПСшВ▓хТМц▒ЙшпнуАВ[0m
